lib/explorer/data_frame.ex

```diff
@@ -4721,6 +4721,51 @@ defmodule Explorer.DataFrame do
     %{out_df | groups: Enum.filter(df.groups, &(&1 in id_columns))}
   end
 
+  @doc """
+  Add a row index as the first column in the DataFrame.
+
+  The resulting column does not have any special properties. It is a regular
+  column of type UInt32.
+
+  ## Options
+
+    * `:name` - Name of the column to add.
+    * `:offset` - Start the row count at this offset. Defaults to 0.
+
+  ## Examples
+
+      iex> df = Explorer.DataFrame.new(a: [1, 3, 5], b: [2, 4, 6])
+      iex> Explorer.DataFrame.with_row_count(df)
+      #Explorer.DataFrame<
+        Polars[3 x 3]
+        row_nr u32 [0, 1, 2]
+        a s64 [1, 3, 5]
+        b s64 [2, 4, 6]
+      >
+
+      iex> df = Explorer.DataFrame.new(a: [1, 3, 5], b: [2, 4, 6])
+      iex> Explorer.DataFrame.with_row_count(df, name: "id", offset: 1000)
+      #Explorer.DataFrame<
+        Polars[3 x 3]
+        id u32 [1000, 1001, 1002]
+        a s64 [1, 3, 5]
+        b s64 [2, 4, 6]
+      >
+
+  """
+  @doc type: :single
+  @spec with_row_count(df :: DataFrame.t(), opts :: Keyword.t()) :: DataFrame.t()
+  def with_row_count(%DataFrame{} = df, opts \\ []) do
```

Comment:
Not sure if this should be `def with_row_count(%DataFrame{} = df, column_name, index)`